### DISCUSSION TITLE
State of hydrogen git repository tags and branches?

### DISCUSSION THREAD

**Discussion by SpotlightKid** [hydrogen-music/hydrogen#2075] [Q&A] [answered]
Hello,

Can somebody explain to me the state of the hydrogen git repo?

* The tag for the latest release 1.2.4 does not exist on the master branch.
* When I look [at the tag](https://github.com/hydrogen-music/hydrogen/releases/tag/1.2.4) it says there are 1334 commits to master since this release:
* The 1.2.4 tag exists on the releases/1.2 branch, but when I look at this branch it says it's 698 commits ahead of, 1334 commits behind master.

Yet all the documentation for building hydrogen says to just clone the repo and doesn't mention switching to any other branch than the default (master).

Can somebody shed some light on why this is so and which is the correct branch to build from if you want the latest development state?

**Comment by theGreatWhiteShark** [comment, marked as answer]
Hey @SpotlightKid ,

> The tag for the latest release 1.2.4 does not exist on the master branch.

Nope. The 1.2.X release line is kept in branch `releases/1.2`. `master` contains the cutting-edge stable development code for a future version 2.0.

> The 1.2.4 tag exists on the releases/1.2 branch, but when I look at this branch it says it's 698 commits ahead of, 1334 commits behind master.

Yeah, some massive changes needed to be done (and there are more to come) for the features upcoming in 2.0.

> Yet all the documentation for building hydrogen says to just clone the repo and doesn't mention switching to any other branch than the default (master).

Hmmm... Well, I don't know whether this is actually a bad thing. I would like people to checkout the latest development on `master` and give feedback, of course. I'll do my best to keep everything stable and the branch _should_ be save to use. Also, I would expect users who compile from source to understand the different between building from HEAD on `master` with respect to building from the latest tagged commit.

**Comment by theGreatWhiteShark** [comment]
In addition,

> The 1.2.4 tag exists on the releases/1.2 branch, but when I look at this branch it says it's 698 commits ahead of, 1334 commits behind master.

this sounds more disturbing than it actually is. About 98% of those `698` commits have been cherry-picked and live in both branches.

**Comment by SpotlightKid** [comment]
Thanks for the explanation.

I guess that leaves me with the decision whether to build the [hydrogen-git AUR package](https://aur.archlinux.org/packages/hydrogen-git) from the master or the releases/1.2 branch, depending on the main purpose:

* master: follow latest developments
* releases/1.2: bug triage and issue testing

I think for now I'll go with the releases/1.2 branch, since (from my POV) version 2.0 still seems far off.

**Reply by theGreatWhiteShark** [reply]
> I think for now I'll go with the releases/1.2 branch, since (from my POV) version 2.0 still seems far off.

Yes. Definitely.

Since changes will be quite profound, I'll do both alpha and beta releases for 2.0 first. Maybe at one of these the branch could be changed back from `releases/1.2` to `master`.

**Comment by SpotlightKid** [comment]
One more thing: I don't know how official this document is, but it might be prudent to update

https://github.com/hydrogen-music/hydrogen/wiki/Packaging-for-LINUX

and mention that release packages (in absence of a source distribution archive for a release) should be built from the release tags, which are on the `releases/1.2` branch and not from the `master` branch.

**Reply by theGreatWhiteShark** [reply]
Thanks for pointing this out! Totally forgot about it.
